### PR TITLE
Upgrade Angular from 16 to 17

### PR DESCRIPTION
**Pull request by fjaguado**
# PR Details

Update to Angular 17.0.9 and TypeScript 5.2.2

## Description

Followed Angular Update Guide

https://update.angular.io/?l=3&v=16.0-17.0

## Related Issue
#130 

## Motivation and Context

We want to have the latest technology.

## How has this been tested

It has been executed tests

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the CONTRIBUTING document
- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation

### CODE REVIEW

File: projects/systelab-translate/src/lib/decimal-format/decimal-format.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ export class DecimalFormat {
 			// toFixed method has bugs on IE (0.7 --> 0)
 			fracStr = num.toString(10)
 				.substr(2);
-			var c = (num >= 1) ? 1 : 0; //carry
+			var c = (Number(num) >= 1) ? 1 : 0; //carry
```
Comment (OscarDeDios) [inline projects/systelab-translate/src/lib/decimal-format/decimal-format.ts:114]: Why is it needed this conversion? num is not implicy number? something new in new typescript version?

Comment (fjaguado) [reply projects/systelab-translate/src/lib/decimal-format/decimal-format.ts:114]: It's a lint new error. num is not implicy number.

**Review by RubKrmon — APPROVED** [review]
Good work

**Review by OscarDeDios — APPROVED** [review]

File: projects/systelab-translate/package.json [hunk 2]
```diff
@@ -19,8 +19,8 @@
   },
   "private": false,
   "peerDependencies": {
-    "@angular/common": "^16.0.0",
-    "@angular/core": "^16.0.0",
+    "@angular/common": "^17.0.0",
+    "@angular/core": "^17.0.0",
     "@ngx-translate/core": "^15.0.0",
     "@ngx-translate/http-loader": "^8.0.0",
     "date-fns": "^2.30.0"
```
Comment (joaquimvila) [inline projects/systelab-translate/package.json:26]: We should consider to upgrade date-fns to version 3.x. Last version is 3.20, so we should be able to use the latest one. Let's evaluate impact and if it's too much, let's add a new ticket to do it separatedly (i.e maybe as part of next PI upgrade),

**Review by joaquimvila — APPROVED** [review]
Just a minor comment about date-fns upgrade

File: projects/systelab-translate/package.json (hunk 2 above)
Comment (fjaguado) [reply projects/systelab-translate/package.json:26]: I tried to update it but there are a lot of changes to be made and a lot of errors regarding that. I'll be creating a new issue to update it

**Pull request merged**